### PR TITLE
Require warn() before liquidate() to mitigate manipulation risk

### PR DESCRIPTION
**Pull request by haydenshively**

### CODE REVIEW

File: core/src/Borrower.sol [hunk 1]
```diff
@@ -88,6 +91,23 @@ contract Borrower is IUniswapV3MintCallback {
                            MAIN ENTRY POINTS
     //////////////////////////////////////////////////////////////*/
 
+    function warn() external {
```
Comment (aneesh1) [inline core/src/Borrower.sol:94]: ~~Want to emit an event?
`event insolvencyWarning (address unhealthyBorrower, uint256 amount0ToRepay, uint256 amount1ToRepay)`~~

~~`emit insolvencyWarning()`~~

Disregard - just saw this is addressed by https://github.com/aloelabs/aloe-ii/pull/48.

**Review by aneesh1 — APPROVED** [review]
Good - prior to calling `account.liquidate()` now you have to call `account.warn()` - should be a nice feature for users.

File: core/src/Borrower.sol (hunk 1 above)
Comment (haydenshively) [reply core/src/Borrower.sol:94]: Next PR :)

**Comment by haydenshively** [comment]
> Good - prior to calling `account.liquidate()` now you have to call `account.warn()` - should be a nice feature for users.

Note that this is only true for the branches of `liquidate` that trigger a swap callback and a 5% bonus. For the simpler liquidation (where we're able to repay everything in-kind without swapping) no warning is required.

**Pull request merged**